lib/ethclient/ethclient.go

```diff
@@ -92,6 +93,25 @@ func (w Wrapper) Address() string {
 	return w.address
 }
 
+// ProgressIfSyncing returns the sync progress (and true) if the node is syncing else false if node is
+// not syncing or an error.
+//
+// This wrap SyncProgress which returns nil-nil if the node is not syncing which results in panics.
+func (w Wrapper) ProgressIfSyncing(ctx context.Context) (*ethereum.SyncProgress, bool, error) {
```

Comment:
Would it be simpler to assume that every client is _always_ syncing with the only question is how much it is behind. In this case we could always return the sync progress struct, but for nodes that are not behind, the `CurrentBlock == HighestBlock` would always hold, which could be quickly checked with something like `Done()` (or a new function `Synced` meaning we're on the tip, because we are never really "done") 

(In fact this is exactly how I initally understood this API and overlooked the pointer)

Reply:
I feel that would change the API too much, I'd rather stick with standard API, just make it safer.